go/src/meguca/geoip/db.go

```diff
@@ -45,9 +101,14 @@ func LookUp(ip string) (iso string) {
 			ISOCode string `maxminddb:"iso_code"`
 		} `maxminddb:"country"`
 	}
-	if err := db.Lookup(dec, &record); err != nil {
+
+	rw.Lock()
```

Comment:
What is this for?

Reply:
The mutex?
For the geoip db, like you asked.

Reply:
Ah, it is a lookup, do we know if that edits any data?
We can use RLock and RUnlock if it doesn't.

Reply:
You only need the lock for the pointer switching, so that is not required.

Reply:
I'll remove it, then.
Should I get rid of the explicit locks and use defer, or is it aesthetically acceptable?